corehq/apps/users/models.py

```diff
@@ -531,14 +531,17 @@ def add_domain_membership(self, domain, timezone=None, **kwargs):
         self.domain_memberships.append(domain_membership)
         self.domains.append(domain)
 
-    def add_as_web_user(self, domain, role, location_id=None, program_id=None):
+    def add_as_web_user(self, domain, role, primary_location_id=None,
+                        assigned_locations_ids=[], program_id=None):
```

Comment:
I think it's best practice to not have mutable objects as default values https://docs.python.org/3/reference/compound_stmts.html#function-definitions

> This is especially important to understand when a default parameter value is a mutable object, such as a list or a dictionary: if the function modifies the object (e.g. by appending an item to a list), the default parameter value is in effect modified. This is generally not what was intended. A way around this is to use None as the default, and explicitly test for it in the body of the function, e.g.:

So should assign default value as none and in the body do 
```
if assigned_locations_ids is None:
    assigned_locations_id = []
```